packages/fast-components-styles-msft/src/badge/index.ts

```diff
@@ -0,0 +1,108 @@
+import {
+    ComponentStyles,
+    ComponentStyleSheet,
+    CSSRules,
+} from "@microsoft/fast-jss-manager";
+import { BadgeClassNameContract } from "@microsoft/fast-components-class-name-contracts-msft";
+import {
+    applyLocalizedProperty,
+    Direction,
+    ellipsis,
+    toPx,
+} from "@microsoft/fast-jss-utilities";
+import { DesignSystem, withDesignSystemDefaults } from "../design-system/index";
+import { ensureNormalContrast } from "../utilities/colors";
+import { fontWeight } from "../utilities/fonts";
+
+function smallBadgeStyle(direction: Direction): CSSRules<DesignSystem> {
+    return {
+        padding: direction === Direction.ltr ? "0 7px 2px 0" : "0 0 2px 7px",
+        height: "14px",
+        lineHeight: "16px",
+        "&$badge__highlight, &$badge__lowlight, &$badge__accent": {
+            padding: "0 7px 4px",
+        },
+    };
+}
+
+function largeBadgeStyle(direction: Direction): CSSRules<DesignSystem> {
+    return {
+        [applyLocalizedProperty("paddingRight", "paddingLeft", direction)]: "14px",
+        height: "20px",
+        "&$badge__highlight, &$badge__lowlight, &$badge__accent": {
+            padding: "2px 14px 4px",
+        },
+    };
+}
+
+function backplateStyle(designSystem: DesignSystem): CSSRules<DesignSystem> {
+    return {
+        borderRadius: toPx(designSystem.cornerRadius),
+    };
+}
+
+const styles: ComponentStyles<BadgeClassNameContract, DesignSystem> = (
+    config: DesignSystem
+): ComponentStyleSheet<BadgeClassNameContract, DesignSystem> => {
+    const designSystem: DesignSystem = withDesignSystemDefaults(config);
+    const direction: Direction = designSystem.direction;
+    const contrast: number = designSystem.contrast;
+    // Badges do not switch color on theme change
+    const foregroundColor: string = "#111";
+    const backgroundColor: string = "#FFF";
+    const lowlightBackground: string = "#333";
+    const highlightBackground: string = "#FFD800";
+    const accentBackground: string = designSystem.brandColor;
+    const accentForegroundColor: string = ensureNormalContrast(
+        contrast,
+        foregroundColor,
+        accentBackground
+    );
+    const hightlightForegroundColor: string = ensureNormalContrast(
+        contrast,
+        foregroundColor,
+        highlightBackground
+    );
+    const lowlightForegroundColor: string = ensureNormalContrast(
+        contrast,
+        backgroundColor,
+        lowlightBackground
+    );
+
+    return {
+        badge: {
+            fontSize: "13px",
+            lineHeight: "20px",
+            ...ellipsis(),
```

Comment:
I see that we're adding ellipsis; assuming we need maxlines to ensure a single line of text as well?

Reply:
No, it will never go over one line, but without a max-width, the ellipsis will never show... this will just be one line indefinitely. The old version did not have a max-width. Perhaps this one should.

Reply:
I put in a max width of `350px` for now... design can weigh-in on final thoughts.

Reply:
350px is pretty wide, but that's fine. I can't think of a good use for this component with lots of text though, so we should never hit this case.

Reply:
Since this is just style, it wouldn't be a breaking change to alter it in the future. Perhaps we could go a bit smaller and see how it works out. The entire point of the having the ellipses is that they actually come into play. Maybe 200 or 225?

Reply:
Went with `215px` - seems pretty good for now.